include/swift/AST/Expr.h

```diff
@@ -3853,15 +3862,66 @@ class OpaqueValueExpr : public Expr {
 
 public:
   explicit OpaqueValueExpr(SourceRange Range, Type Ty,
-                           bool isPlaceholder = false)
+                           bool canBeCaptured = false, bool canBeReused = false)
       : Expr(ExprKind::OpaqueValue, /*Implicit=*/true, Ty), Range(Range) {
-    Bits.OpaqueValueExpr.IsPlaceholder = isPlaceholder;
+    Bits.OpaqueValueExpr.CanBeCaptured = canBeCaptured;
+    Bits.OpaqueValueExpr.CanBeReused = canBeReused;
   }
 
-  /// Whether this opaque value expression represents a placeholder that
-  /// is injected before type checking to act as a placeholder for some
-  /// value to be specified later.
-  bool isPlaceholder() const { return Bits.OpaqueValueExpr.IsPlaceholder; }
+  /// Whether the opaque value can be captured by a closure. Most opaque values
+  /// are positioned relative to their owner such that a closure cannot be
+  /// introduced between them.
+  bool canBeCaptured() const {
+    return Bits.OpaqueValueExpr.CanBeCaptured;
+  }
+
+  /// Whether the opaque value can only appear in one location in the AST or in
+  /// several.
+  bool canBeReused() const {
```

Comment:
Should this be used somewhere in SILGen, or is it only for validation?

Reply:
I only ended up needing it in validation.